typescript-carousel/main.ts

```diff
@@ -0,0 +1,78 @@
+const imgRelativeUrls = [
+  './images/001.png',
+  './images/004.png',
+  './images/007.png',
+  './images/025.png',
+  './images/039.png',
+];
+const $indicators = document.querySelector('#indicators');
+if (!$indicators) throw new Error('$indicators not found!');
+const indicatorList = Array.from(
+  $indicators.querySelectorAll('i')
+) as HTMLElement[];
+const $currentImg = document.querySelector('#current-img') as HTMLImageElement;
+const $mainRow = document.querySelector('#main-row');
+
+let position = 1;
+
+function updateImg(): void {
+  if (!$currentImg) throw new Error('$currentImg not found!');
+  $currentImg.src = imgRelativeUrls[position - 1];
+  for (const indicator of indicatorList) {
+    if (Number(indicator.dataset.pos) === position) {
+      indicator.className = 'fa-solid fa-circle';
+    } else {
+      indicator.className = 'fa-regular fa-circle';
+    }
+  }
+}
+
+function updateCarousel(): void {
+  if (position < 5) {
+    position++;
+  } else {
+    position = 1;
+  }
+  updateImg();
+}
+
+if (!$mainRow) throw new Error('$mainRow not found!');
+$mainRow.addEventListener('click', handleMainRowClick);
+
+function handleMainRowClick(event: Event): void {
+  const eventTarget = event.target as HTMLElement;
+  if (!eventTarget) throw new Error('eventTarget not found!');
+  if (eventTarget.matches('.fa-chevron-left')) {
+    if (position > 1) {
+      position--;
+    } else {
+      position = 5;
+    }
+    updateImg();
+    clearInterval(carouselSwitcher);
+    carouselSwitcher = setInterval(updateCarousel, 3000);
+  } else if (eventTarget.matches('.fa-chevron-right')) {
+    if (position < 5) {
+      position++;
+    } else {
+      position = 1;
+    }
+    updateImg();
+    clearInterval(carouselSwitcher);
+    carouselSwitcher = setInterval(updateCarousel, 3000);
+  }
```

Comment:
I would maybe consider having these in different functions and using one for a `handleLeftClick` function and one for `handleRightClick` function and attach them to the different left and right click icons. This feels like a more involved check but also you are using `else/if` which makes me think that there _should_ be some sort of code at the end but we only need the two options. If we separate them it may look like more code but it will be clearer for what code is doing what job.